examples/run/cuda/seeding_example_cuda.cpp

```diff
@@ -221,21 +223,27 @@ int seq_run(const std::string& detector_file, const std::string& hits_dir,
 // The main routine
 //
 int main(int argc, char* argv[]) {
-    if (argc < 4) {
-        std::cout << "Not enough arguments, minimum requirement: " << std::endl;
-        std::cout << "./seq_example <detector_file> <hit_directory> "
-                     "<events> <skip_cpu>"
-                  << std::endl;
-        return -1;
-    }
-
-    auto detector_file = std::string(argv[1]);
-    auto hit_directory = std::string(argv[2]);
-    auto events = std::atoi(argv[3]);
-    bool skip_cpu = std::atoi(argv[4]);
-
-    std::cout << "Running ./seeding_example " << detector_file << " "
-              << hit_directory << " "
-              << " " << events << " " << skip_cpu << std::endl;
-    return seq_run(detector_file, hit_directory, events, skip_cpu);
+    po::options_description desc("Allowed options");
+    desc.add_options()("detector_file", po::value<std::string>()->required(),
+                       "specify detector file");
+    desc.add_options()("hit_directory", po::value<std::string>()->required(),
+                       "specify the directory of hit files");
+    desc.add_options()("events", po::value<int>()->required(),
+                       "number of events");
+    desc.add_options()("run_cpu", po::value<bool>()->default_value(false),
+                       "run cpu tracking as well");
+
+    po::variables_map vm;
+    po::store(po::parse_command_line(argc, argv, desc), vm);
+    po::notify(vm);
+
+    auto detector_file = vm["detector_file"].as<std::string>();
+    auto hit_directory = vm["hit_directory"].as<std::string>();
+    auto events = vm["events"].as<int>();
+    auto run_cpu = vm["run_cpu"].as<bool>();
+
+    std::cout << "Running ./traccc_seeding_example_cuda " << detector_file
```

Comment:
As long as we're doing this update... Could this printout make use of `argv[0]` instead? For a long time now we've been having these applications print wrong names for themselves...

Reply:
Right. I changed them